arangod/Graph/BaseOptions.cpp

```diff
@@ -614,3 +607,53 @@ void BaseOptions::isQueryKilledCallback() const {
     THROW_ARANGO_EXCEPTION(TRI_ERROR_QUERY_KILLED);
   }
 }
+
+void BaseOptions::setVertexProjections(Projections projections) {
+  _vertexProjections = std::move(projections);
+}
+
+void BaseOptions::setEdgeProjections(Projections projections) {
+  _edgeProjections = std::move(projections);
+}
+
+Projections const& BaseOptions::getVertexProjections() const {
+  return _vertexProjections;
+}
+
+Projections const& BaseOptions::getEdgeProjections() const {
+  return _edgeProjections;
+}
+
+void BaseOptions::toVelocyPackBase(VPackBuilder& builder) const {
+  TRI_ASSERT(builder.isOpenObject());
+  builder.add("parallelism", VPackValue(_parallelism));
+  builder.add(StaticStrings::GraphRefactorFlag, VPackValue(refactor()));
+  builder.add("produceVertices", VPackValue(_produceVertices));
+
+  if (!_vertexProjections.empty()) {
+    _vertexProjections.toVelocyPack(builder, "vertexProjections");
+  }
+
+  if (!_edgeProjections.empty()) {
+    _edgeProjections.toVelocyPack(builder, "edgeProjections");
+  }
+}
+
+void BaseOptions::parseShardIndependentFlags(arangodb::velocypack::Slice info) {
+  // parallelism is optional
+  _parallelism = VPackHelper::getNumericValue<size_t>(info, "parallelism", 1);
+
+  TRI_ASSERT(_produceVertices);
+  _produceVertices =
+      VPackHelper::getBooleanValue(info, "produceVertices", true);
+
+  // TODO: For some reason _produceEdges has not been deserialized (skipped
+  // in refactoring to avoid side-effects)
+
```

Comment:
Is this an actual TODO?

Reply:
Yes, but unrelated to this PR.
While writing this code I found this piece, but have not yet analyzed the side-effects.
So I wanted to keep this PR clean of this change.